zk/datastream/server/data_stream_server.go

```diff
@@ -255,6 +255,64 @@ func (srv *DataStreamServer) CreateAndBuildStreamEntryBytes(
 	return result, nil
 }
 
+func (srv *DataStreamServer) GetHighestBlockNumber() (uint64, error) {
+	header := srv.stream.GetHeader()
+
+	if header.TotalEntries == 0 {
+		return 0, nil
+	}
+
+	//find end block entry to delete from it onward
+	entryNum := header.TotalEntries - 1
+	var err error
+	var entry datastreamer.FileEntry
+	for {
+		entry, err = srv.stream.GetEntry(entryNum)
+		if err != nil {
+			return 0, err
+		}
+		if entry.Type == datastreamer.EntryType(3) {
```

Comment:
FYI this will move to 2 in the new proto format, 3 denotes a transaction in that world